### ISSUE TITLE
Uncaught TypeError: Cannot read property 'id' of null

### ISSUE THREAD

**Issue by thegoldenmule** [ajaxorg/ace#3948]
Hi there, we are having trouble with an exception thrown from very specific usage.

Repro Steps:

1. Go to the editor on https://ace.c9.io.
2. Add the following script: `module.exports = {`. Note the missing brace at the end.

Results:

```
worker-javascript.js:1 Uncaught TypeError: Cannot read property 'id' of null
    at Object.e.nud (worker-javascript.js:1)
    at Q (worker-javascript.js:1)
    at worker-javascript.js:1
    at Object.i.led (worker-javascript.js:1)
    at Q (worker-javascript.js:1)
    at At (worker-javascript.js:1)
    at Ot (worker-javascript.js:1)
    at yn (worker-javascript.js:1)
    at exports.JavaScriptWorker.onUpdate (worker-javascript.js:1)
    at r (worker-javascript.js:1)
```

Additional Notes:

While this seems like an edge case, it's actually occurring frequently with our users because this error is thrown while typing out a `module.exports` statement.

**Comment by nightwing** [comment]
Looks like a bug in the old version of jshint we use, we'll switch to eslint for the next release of ace.

**Comment by tdsang01** [comment]
I'm having the same problem

**Comment by mfiorek** [comment]
The same error occurs when typing `return {` in the last line of the editor. I believe this may occur quite often for many users.

**Comment by sleepomeno** [comment]
@nightwing is there an expected date for that new release using eslint where that exception doesn't occur any more?

**Comment by nightwing** [comment]
This is already fixed in ace v1.4.12. I don't see the original error on https://ace.c9.io/
